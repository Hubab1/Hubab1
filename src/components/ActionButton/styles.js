import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
    height: 45px;
`;

export const label = css`
    text-transform: none;
    font-size: 16px;
`;

export const mainDisabled = css`
    background-color: #D5D8DE !important;
    color: #ffffff !important;
    box-shadow: 0px 1px 5px 0px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 3px 1px -2px rgba(0,0,0,0.12) !important;
`;

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0,
    textDecoration: 'none',
    display: 'block',
}));
