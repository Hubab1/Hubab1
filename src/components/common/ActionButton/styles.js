import {
    css
} from 'emotion';
import styled from '@emotion/styled';

export const root = css`
    border-radius: 21.5px !important;
`

export const label = css`
    text-transform: none;
`

export const ButtonContainer = styled.div(props => ({
    marginTop: props.marginTop || 0,
    marginBottom: props.marginBottom || 0
}))
