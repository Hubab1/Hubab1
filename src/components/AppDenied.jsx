import React from 'react';
import styled from '@emotion/styled';
import { css } from 'emotion';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { ROUTES } from 'app/constants';
import withRelativeRoutes from 'app/withRelativeRoutes';
import { H1, SpacedH3 } from 'assets/styles';
import cry from 'assets/images/cry.svg';
import ActionButton from 'components/common/ActionButton/ActionButton';
import clsx from 'clsx';
import DenialReason from 'components/AppDenialReason';
import API from 'app/api';

export const Img = styled.img`
    padding-top: 10px;
    height: 200px;
`;

export const applicationUnit = css`
  color: #454B57;
  font-size: 14px;
  line-height: 17px;
  text-align: center;
  padding-top: 10px;
`;

export class AppDenied extends React.Component {
    state = {
        viewDenialReason: false,
    };

    toggleViewDenialDecision = () => {
        this.setState({viewDenialReason: !this.state.viewDenialReason});
    };

    getDenialDecisionDate(date) {
        const d = new Date(date);
        return d.toLocaleDateString('en-US');
    }

    render () {
        const { profile, configuration, applicant } = this.props;
        if (!profile || !configuration || !applicant) return null;

        const { viewDenialReason } = this.state;
        const {
            unit,
            last_status_change,
        } = profile;

        const buildingName = configuration.community.building_name || configuration.community.normalized_street_address;
        const unitNumber = (!!unit && !!unit.unit_number) ? ` Unit ${unit.unit_number}` : '';
        const denialDecisionDate = this.getDenialDecisionDate(last_status_change.created_at);
        const { name } = applicant.client.person;

        return (
            <>
                <div className={clsx({'hide-element': viewDenialReason})}>
                    <H1>Application Denied</H1>
                    <SpacedH3>Unfortunately, we were unable to approve your application.</SpacedH3>
                    <Img src={cry}/>
                    <div id="application-unit" className={applicationUnit}>{buildingName}{unitNumber}</div>
                    <ActionButton
                        marginTop={150}
                        onClick={this.toggleViewDenialDecision}
                    >Learn Why</ActionButton>
                </div>
                {viewDenialReason &&
                    <DenialReason
                        date={denialDecisionDate}
                        buildingName={buildingName}
                        unitNumber={unitNumber}
                        name={name}
                        onAgree={this.toggleViewDenialDecision}
                    />
                }
            </>
        )
    }
}
AppDenied.propTypes = {
    profile: PropTypes.object,
    configuration: PropTypes.object,
    applicant: PropTypes.object,
};
const mapStateToProps = state => ({
    profile: state.renterProfile,
    configuration: state.configuration,
    applicant: state.applicant,
});
export default connect(mapStateToProps)(withRelativeRoutes(AppDenied, ROUTES.APP_DENIED));
