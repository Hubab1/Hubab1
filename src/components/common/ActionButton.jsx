import React from 'react';

import Button from '@material-ui/core/Button';
import { root, label } from 'assets/emotion/ActionButton';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children } = this.props;
        return (
            <Button
                onClick={onClick}
                classes={{ root, label }}
                variant="contained"
                color={color || 'primary'}
                type="submit"
                disabled={disabled}
                fullWidth
            >
                {children}
            </Button>
        );
    }
}