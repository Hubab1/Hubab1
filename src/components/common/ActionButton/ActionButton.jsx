import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';

import { root, label, ButtonContainer } from './styles';

export default class ActionButton extends React.Component {
    render () {
        const { color, disabled, onClick, children, marginTop, marginBottom, variant } = this.props;
        return (
            <ButtonContainer marginTop={marginTop} marginBottom={marginBottom}>
                <Button
                    onClick={onClick}
                    classes={{ root, label }}
                    variant={variant}
                    color={color}
                    type="submit"
                    disabled={disabled}
                    fullWidth
                >
                    {children}
                </Button>
            </ButtonContainer>
        );
    }
}

ActionButton.propTypes = {
    color: PropTypes.string,
    onClick: PropTypes.func,
    disabled: PropTypes.bool,
    marginTop: PropTypes.string,
    marginBottom: PropTypes.string,
    variant: PropTypes.string,
}

ActionButton.defaultProps = {
    variant: 'contained',
    color: 'primary',
}