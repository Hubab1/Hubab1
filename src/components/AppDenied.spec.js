import React from 'react';
import { shallow } from 'enzyme';
import { AppDenied } from './AppDenied';
import ActionButton from 'components/common/ActionButton/ActionButton';
import API from 'app/api';
import DenialReason from 'components/AppDenialReason';


const buildProps = (buildingName = 'Fake Building', streetAddress = '123 Fake Street', unitNumber = '2B') => {
    return {
        profile: {
            events: [{event: '270'}],
            unit: {
                unit_number: unitNumber,
            },
            last_status_change: {
                created_at: '2020-05-06 04:06:23'
            }
        },
        applicant: {
            client: {
                person: {
                    name: 'John Doe'
                },
            },
        },
        configuration: {
            community: {
                building_name: buildingName,
                normalized_street_address: streetAddress,
            }
        },
    }
};

it('matches snapshot', () => {
    const wrapper = shallow(<AppDenied {...buildProps()}/>);
    expect(wrapper.getElement()).toMatchSnapshot();
});

describe('application unit', () => {
    it('displays building name when no unit', () => {
        const props = buildProps('Fake Building', '123 Fake Street', null);
        props.unit = null;

        const wrapper = shallow(<AppDenied {...props} />);

        expect(wrapper.find('#application-unit').text()).toEqual('Fake Building')
    });

    it('displays building name without unit when no unit number', () => {
        const props = buildProps('Fake Building', '123 Fake Street', null);

        const wrapper = shallow(<AppDenied {...props} />);

        expect(wrapper.find('#application-unit').text()).toEqual('Fake Building')
    });

    it('displays building name with unit when has unit number', () => {
        const props = buildProps('Fake Building', '123 Fake Street', '7F');

        const wrapper = shallow(<AppDenied {...props} />);

        expect(wrapper.find('#application-unit').text()).toEqual('Fake Building Unit 7F')
    });

    it('displays normalized street address and unit number when no building name', () => {
        const props = buildProps(null, '123 Fake Street', '7F');

        const wrapper = shallow(<AppDenied {...props} />);

        expect(wrapper.find('#application-unit').text()).toEqual('123 Fake Street Unit 7F')
    });
});