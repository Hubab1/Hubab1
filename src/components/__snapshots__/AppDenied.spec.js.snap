// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className=""
>
  <ForwardRef(render)>
    Application Denied
  </ForwardRef(render)>
  <ForwardRef(render)>
    Unfortunately, we were unable to approve your application.
  </ForwardRef(render)>
  <ForwardRef(render)
    src="cry.svg"
  />
  <div
    className="css-x0fehc"
    id="application-unit"
  >
    Fake Building
     Unit 2B
  </div>
  <AdverseActionNoticeButton
    componentType="button"
    marginTop={50}
  />
</div>
`;
